src/otx/algo/samplers/balanced_sampler.py

```diff
@@ -0,0 +1,174 @@
+# Copyright (C) 2024 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+"""Balanced sampler for imbalanced data."""
+
+from __future__ import annotations
+
+import math
+from typing import TYPE_CHECKING
+
+import torch
+from datumaro.components.annotation import Annotation, AnnotationType, LabelCategories
+from torch.utils.data import Sampler
+
+if TYPE_CHECKING:
+    from datumaro import Dataset as DmDataset
+
+    from otx.core.data.dataset.base import OTXDataset
+
+
+def get_idx_list_per_classes(dm_dataset: DmDataset) -> dict[str, list[int]]:
+    """Compute class statistics."""
+    labels = dm_dataset.categories().get(AnnotationType.label, LabelCategories())
+
+    def get_label(ann: Annotation) -> str:
+        """Get the name of the label associated with the given annotation."""
+        try:
+            return labels.items[ann.label].name if ann.label is not None else None
+        except IndexError:
+            return ann.label
```

Comment:
```suggestion
        return ann.label
```
Is there any mandatory reason to obtain a string key?

Reply:
Thanks - i change this to use just ann.label - https://github.com/openvinotoolkit/training_extensions/pull/2993/commits/6eb345a7526d42dafb6351fc7de10e41341786b8